### PR TITLE
Fix for https://github.com/nats-io/jetstream/issues/297

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

Resolves https://github.com/nats-io/jetstream/issues/297

/cc @nats-io/core

### CODE REVIEW

File: test/jetstream_test.go [hunk 1]
```diff
@@ -4547,9 +4547,29 @@ func TestJetStreamStreamPurge(t *testing.T) {
...
 			mset.Purge()
-			if state := mset.State(); state.Msgs != 0 {
+			state := mset.State()
+			if state.Msgs != 0 {
 				t.Fatalf("Expected %d messages, got %d", 0, state.Msgs)
 			}
+			// Make sure first timestamp are reset.
+			if !state.FirstTime.IsZero() {
+				t.Fatalf("Expected the state's first time to be zero after purge")
+			}
```
Comment (ripienaar) [inline test/jetstream_test.go:4557]: need to also fix last ts

```
			// Make sure timestamps are reset.
			if !state.FirstTime.IsZero() {
				t.Fatalf("Expected the state's first time to be zero after purge")
			}
			if !state.LastTime.IsZero() {
				t.Fatalf("Expected the state's last time to be zero after purge")
			}
```

Comment (derekcollison) [reply test/jetstream_test.go:4557]: I had it that way but if I reset it I can not tell when the last message was received so reverted.

Comment (ripienaar) [reply test/jetstream_test.go:4557]: hmm, but if its purged its concceptually as if a last message was never received.

Comment (derekcollison) [reply test/jetstream_test.go:4557]: Kindof, we still track next first seq etc and like I said as I was playing with it for myself it seemed like I really wanted to know when the last message had been received regardless of if it had expired or been purged.

**Comment by ripienaar** [comment]
LGTM

**Pull request merged**